html/pfappserver/lib/pfappserver/Form/Config/Source/LDAP.pm

```diff
@@ -0,0 +1,182 @@
+package pfappserver::Form::Config::Source::LDAP;
+
+=head1 NAME
+
+pfappserver::Form::Config::Source::LDAP - Web form for a LDAP user source
+
+=head1 DESCRIPTION
+
+Form definition to create or update a LDAP user source.
+
+=cut
+
+use pf::Authentication::Source::LDAPSource;
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Form::Config::Source';
+with 'pfappserver::Base::Form::Role::Help';
+
+our $META = pf::Authentication::Source::LDAPSource->meta;
+
+
+# Form fields
+has_field 'host' =>
+  (
+   type => 'Text',
+   label => 'Host',
+   element_class => ['input-small'],
+   element_attr => {'placeholder' => '127.0.0.1'},
+   default => $META->get_attribute('host')->default,
+  );
+has_field 'port' =>
+  (
+   type => 'PosInteger',
+   label => 'Port',
+   element_class => ['input-mini'],
+   element_attr => {'placeholder' => '389'},
+   default => $META->get_attribute('port')->default,
+  );
+has_field 'connection_timeout' =>
+  (
+    type         => 'PosInteger',
+    label        => 'Connection timeout',
+    element_attr => {
+        'placeholder' => $META->get_attribute('connection_timeout')->default
+    },
+    default => $META->get_attribute('connection_timeout')->default,
+  );
+has_field 'encryption' =>
+  (
+   type => 'Select',
+   label => 'Encryption',
+   options =>
+   [
+    { value => 'none', label => 'None' },
+    { value => 'ssl', label => 'SSL' },
+    { value => 'starttls', label => 'Start TLS' },
+   ],
+   required => 1,
+   element_class => ['input-small'],
+   default => 'none',
+  );
+has_field 'basedn' =>
+  (
+   type => 'Text',
+   label => 'Base DN',
+   required => 1,
+   default => '',
```

Comment:
The comment does not know what he is talking about for he does not know why that line exists